### PR TITLE
v4l2Render: fix some issues found by static scan tool

### PR DESCRIPTION
**Pull request by wudping**
1, define the base class's destructor as virtual.
2, make sure the pointer is not NULL before accessing.

to fix static code scan problems.

Signed-off-by: wudping <dongpingx.wu@intel.com>

### CODE REVIEW

File: tests/v4l2decode.cpp [hunk 1]
```diff
@@ -192,27 +196,27 @@ int main(int argc, char** argv)
     SharedPtr<V4L2Device> device = V4L2Device::Create();
     if (!device) {
         ERROR("failed to create v4l2 device");
-        return -1;
+        DELETE_RETURN(input);
```
Comment (xuguangxin) [inline tests/v4l2decode.cpp]: Can we declare input as SharedPtr<DecodeInput>

**Comment by wudping** [comment]
hi, @xuguangxin , has used shared_ptr. please help to review again. Thanks.

**Comment by wudping** [comment]
issue about tool is fixed, Thanks.

File: tests/v4l2decode.cpp [hunk 2]
```diff
@@ -73,7 +73,7 @@ static uint32_t renderFrameCount = 0;
 
 static DecodeParameter params;
 
-bool feedOneInputFrame(DecodeInput* input, const SharedPtr<V4L2Device>& device, int index = -1 /* if index is not -1, simple enque it*/)
+bool feedOneInputFrame(SharedPtr<DecodeInput> input, const SharedPtr<V4L2Device>& device, int index = -1 /* if index is not -1, simple enque it*/)
```
Comment (xuguangxin) [inline tests/v4l2decode.cpp]: Please change to `const SharedPtr<DecodeInput>& input`

**Pull request closed without merging**